### PR TITLE
Add RainPoint Smart Water Timer

### PR DESCRIPTION
**Pull request by josiasmontag**
Adds the RainPoint Smart Water Timer (TTV103FRF)

image

image

### CODE REVIEW

File: custom_components/tuya_local/devices/rainpoint_water_timer.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          - dps_val: "1"
+            value: "Manual"
+          - dps_val: "2"
+            value: "Auto"
+          - dps_val: "3"
+            value: "Rain Delay"
+  - entity: binary_sensor
+    category: diagnostic
+    class: problem
+    icon: "mdi:alert"
```
Comment (make-all) [inline custom_components/tuya_local/devices/rainpoint_water_timer.yaml:108]: problem sensor does not need an icon defined, the base class defines dynamic icons already

File: custom_components/tuya_local/devices/rainpoint_water_timer.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        optional: true
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+            value: false
+          - dps_val: false
+            value: true
+  - entity: select
+    name: Weekly Irrigation Rain Delay
```
Comment (make-all) [inline custom_components/tuya_local/devices/rainpoint_water_timer.yaml:34]: This name is a bit long, other similar devices use just "Rain delay".   Also, naming needs to follow HA naming convention.. https://developers.home-assistant.io/docs/core/entity/#:~:text=Entity%20names%20should%20start%20with,a%20capitalized%20abbreviation%20of%20course).

File: custom_components/tuya_local/devices/rainpoint_water_timer.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
+name: RainPoint Smart Water Timer
```
Comment (make-all) [inline custom_components/tuya_local/devices/rainpoint_water_timer.yaml:1]: This top level name should be without branding (and following HA naming convention, see below).

Just "Water timer" should be enough.

**Comment by josiasmontag** [comment]
@make-all Updated, please check

**Comment by MariusNovak** [comment]
Any idea how to add the RainPoint Smart Water Timer Twin way / two zone TWG004WRF ?
using this using this custom component alows only to use one of the two zones ... 
Thanks for adive

**Pull request merged**